### PR TITLE
feat: EPUB & XHTML export routine, template handling improvement and code cleaning.

### PR DESCRIPTION
**Pull request by SteelWagstaff**
This PR is a partial fix for https://github.com/pressbooks/pressbooks/issues/2209 and https://github.com/pressbooks/pressbooks/issues/2490. It refactors and simplifies our EPUB and XHTML export routines regarding to the template handling. 

For EPUB this also modifies the routine, so that Pressbooks consistently produces valid EPUB following 3.2 specs and retires the previous EPUB 2.01 and EPUB 3 export options.

Some style changes are needed, those are addressed in pressbooks-book repo and probably other themes. Those changes TK. We prefer to wait for those changes before merge this one.

### CODE REVIEW

File: inc/modules/export/namespace.php [hunk 1]
```diff
@@ -283,6 +275,34 @@ function update_pins() {
...
+		$buffer .= "<dc:{$type} id='{$type}{$contributor_number}'>{$contributor}</dc:{$type}>\n";
+		$buffer .= "<meta refines='#{$type}{$contributor_number}' property='role' scheme='marc:relators'>{$role}</meta>\n";
+		// TODO: add file-as if possible
+		// echo "<meta refines='#creator{$contributor_number}' property='file-as'>last name, first name</meta>";
+		$index++;
+	}
+
+	return $buffer;
+}
+
```
Comment (fdalcin) [inline inc/modules/export/namespace.php:305]: This whole block could be removed since

**Review by fdalcin — APPROVED** [review]
LGTM folks!

**Review by SteelWagstaff — COMMENTED** [review]
@pressbooks/developers Footnote content is not being rendered in front and back matter in experimental Pressbooks

File: inc/modules/export/xhtml/class-xhtml11.php [hunk 2]
```diff
@@ -1247,39 +1247,44 @@ protected function renderFrontMatterGenerator( $book_contents, $metadata ) : \Ge
...
+				'epigraph',
+				'title-page',
+				'before-title',
+			];
 
-			if ( 'dedication' === $subclass || 'epigraph' === $subclass || 'title-page' === $subclass || 'before-title' === $subclass ) {
-				continue; // Skip
+			if ( in_array( $data['subclass'], $skipClasses, true ) ) {
+				continue;
 			}
```
Comment (fdalcin) [inline inc/modules/export/xhtml/class-xhtml11.php]: This is a minor improvement to the conditional

**Review by fdalcin — APPROVED** [review]

**Pull request merged**